### PR TITLE
docs: inventory updates

### PR DESCRIPTION
**Pull request by aldeed**
Impact: **minor**  
Type: **docs**

New and updates docs articles to reflect changes from PRs https://github.com/reactioncommerce/reaction/pull/5164 and https://github.com/reactioncommerce/reaction/pull/5186

- Added "Inventory" article to "Concepts" section
- Added "Inventory Basics for Developers" article, which explains the way the core inventory plugin works for Reaction developers
- Added "Simple Inventory Plugin" article for devs who want to understand the code and features provided by that plugin
- Added "How To: Create an Inventory Data Provider Plugin" for devs who need to replace the Simple Inventory plugin with a custom one
- Added a new task to the storefront UI guide, "Show inventory status badges", which explains how to use inventory fields to show inventory status in the storefront UI
- Various other small updates related to inventory changes

### CODE REVIEW

File: public-docs/concepts-inventory.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+title: Inventory
```
Comment (kieckhafer) [inline public-docs/concepts-inventory.md]: Does this need an `id` too?

File: public-docs/core-plugins-simple-inventory.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- Adds an "Inventory" card to the product variant editor UI on the operator site. This card shows and updates inventory info using the aforementioned GraphQL mutation and query.
+- Listens for order status changes to track the current "reserved" inventory quantity.
+- Provides a `recalculateReservedSimpleInventory` mutation and a UI button for it. This recalculates the "reserved" inventory quantity if it becomes incorrect.
+- Allows you to set a `lowInventoryWarningThreshold` number. When the available inventory (in stock minus reserved) is at or below this threshold, `isLowQuantity` will be `true` when inventory info is queried.
+
+## When to update inventory
+
+The Simple Inventory plugin considers variant inventory to be reserved starting when an order is placed for that variant, and ending when all payments for that order have been approved. You should notice that approving the final payment will automatically decrease the reserved number, decrease the "in stock" number, and keep the "available" number the same.
```
Comment (kieckhafer) [inline public-docs/core-plugins-simple-inventory.md]: Quote "reserved" to keep it consistant with "in stock" and "available" here.

File: public-docs/core-plugins-simple-inventory.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Similarly, if you cancel an order or some items from an order, you should see the reserved number decrease. And if you cancel after already approving all payments, you are asked if you want to increase the "in stock" number for all canceled items.
```
Comment (kieckhafer) [inline public-docs/core-plugins-simple-inventory.md]: Same as above, quote "reserved"

File: website/versioned_docs/version-2.0.0/concepts-inventory.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Most inventory plugins will want to track new orders and consider the ordered quantity of each item to be "reserved" until you pull the order and decrease the "in stock" quantity. If an inventory plugin doesn't track this, then "reserved" inventory will always be zero.
+
+### Inventory Available To Sell
+
+The "available to sell" number represents how many units the system will allow a shopper to order. It is calculated by subtracting "reserved" from "in stock". If an inventory plugin doesn't track "reserved", then the available inventory will always be the same as the in stock inventory.
+
+### Can Backorder
+
+Each variant may or may not have backordering enabled. With backordering enabled, the orders service will allow an order to be placed for a quantity higher than the "available to sell" quantity.
```
Comment (kieckhafer) [inline website/versioned_docs/version-2.0.0/concepts-inventory.md]: Inconsistant use of hyphens in the word `backordering`. I'm not sure which is correct, but we should stick to one.

File: website/versioned_docs/version-2.0.0/core-plugins-simple-inventory.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+These are all situations in which you likely want to have no known inventory quantity but yet allow ordering any quantity. In other cases, though, you may be synchronizing inventory data from an external system, and a variant with no tracked inventory might mean it hasn't synced yet and therefore should be considered sold out (and ideally also invisible).
+
+To accommodate either situation, the core "inventory" plugin has a shop setting, found on the Shop Settings page of the operator UI. The setting, `canSellVariantWithoutInventory`, determines whether a variant with no inventory information can be sold. This setting is `true` by default.
+- If you leave it `true`, then you will need to enable inventory tracking, disable back-ordering, and set "in stock" inventory to `0` for each variant that you want to appear sold out.
+- If you change it to `false`, then for products that should allow unlimited sales, you have two options:
+    - Enable inventory tracking, set "in stock" inventory to `0` and enable back-ordering, or
```
Comment (kieckhafer) [inline website/versioned_docs/version-2.0.0/core-plugins-simple-inventory.md]: Inconsistant use of hyphens in the word `back-ordering`. I'm not sure which is correct, but we should stick to one.

File: website/versioned_docs/version-2.0.0/how-to-create-inventory-plugin.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+- If you use a well known third-party inventory system, there may be a community plugin already available for it. If not, consider making yours available as open source so that the community can help you maintain it.
+
+Once you're sure you need to create a custom inventory plugin, use the standard steps to create the plugin boilerplate. Beyond that, an inventory plugin needs to do the following:
+- Register an `inventoryForProductConfigurations` type function that returns current inventory in stock and other related data for a list of product configurations.
+- Optionally provide a way of editing the inventory in stock in the operator UI, by registering client components
+- Optionally track order status changes to keep track of "reserved" inventory, which is inventory that is still technically in stock but should not be considered available to sell.
+
+## Register an `inventoryForProductConfigurations` function
```
Comment (kieckhafer) [inline website/versioned_docs/version-2.0.0/how-to-create-inventory-plugin.md]: In the docs above, you state you must add both a `inventoryForProductConfigurations` and `inventoryForProductConfiguration` function, but here you only provide docs for adding a `inventoryForProductConfigurations` with no mention of the other. Should it at least be mentioned that you need to create the other function as well?

Comment (aldeed) [reply website/versioned_docs/version-2.0.0/how-to-create-inventory-plugin.md]: You actually do not need to create both. The core `inventory` plugin has both, but the singular version simply calls the multiple function. It's just for convenience to make it a bit less code when you only need one. I'll explain that here.

File: public-docs/concepts-inventory.md (hunk 1 above)
Comment (aldeed) [reply public-docs/concepts-inventory.md]: Works without it (uses file name) but we do usually include it by convention. I added it.

**Pull request merged**